pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1309,12 +1471,32 @@ func (nc *Controller) cancelPodEviction(node *v1.Node) bool {
 	return false
 }
 
-// evictPods queues an eviction for the provided node name, and returns false if the node is already
-// queued for eviction.
-func (nc *Controller) evictPods(node *v1.Node) bool {
+// evictPods:
+// - If node is not marked as evicted, it will be added to the evictor queue.
+//   The method returns false if the node name was already enqueued, otherwise true.
+// - If node is marked as evicted, pods are deleted immediately.
+//   The method returns false, cause the node wasn't added to the queue.
+func (nc *Controller) evictPods(node *v1.Node) (bool, error) {
 	nc.evictorLock.Lock()
 	defer nc.evictorLock.Unlock()
-	return nc.zonePodEvictor[utilnode.GetZoneKey(node)].Add(node.Name, string(node.UID))
+	status, ok := nc.nodeEvictionMap.getStatus(node.Name)
+	if ok && status == evicted {
+		// Node eviction already happened for this node.
+		// Handling immediate pod deletion.
+		pods, err := nc.getPodsAssignedToNode(node.Name)
```

Comment:
Sorry for going back and forth - I change my mind again. Let's actually try to delete only pods that we believe makes sense. So pass the pods explicitly here to this method.

Reply:
Done.